pkg/pillar/zedUpload/azureutil/azure.go

```diff
@@ -242,3 +244,107 @@ func GetAzureBlobMetaData(accountName, accountKey, containerName, remoteFile str
 	stringHex := hex.EncodeToString(decodedString)
 	return blob.Properties.ContentLength, stringHex, nil
 }
+
+// GenerateBlobSasURI is used to generate the URI which can be used to access the blob until the the URI expries
+func GenerateBlobSasURI(accountName, accountKey, containerName, remoteFile string, httpClient *http.Client, startTime time.Time, endTime time.Time) (string, error) {
+	c, err := NewClient(accountName, accountKey, httpClient)
+	if err != nil {
+		return "", err
+	}
+	blobClient := c.GetBlobService()
+	container := blobClient.GetContainerReference(containerName)
+	containerExists, err := container.Exists()
+	if err != nil {
+		return "", nil
+	}
+	if !containerExists {
+		return "", fmt.Errorf("container doesn't exist")
```

Comment:
Please include the container name also in an error. It will be helpful while debugging.